bundles/go/go_lexer.moon

```diff
@@ -0,0 +1,84 @@
+-- Copyright 2016 The Howl Developers
+-- License: MIT (see LICENSE.md at the top-level directory of the distribution)
+
+howl.aux.lpeg_lexer ->
+  c = capture
+  ident = (alpha + '_')^1 * (alpha + digit + '_')^0
+  ws = c 'whitespace', space
+
+  identifier = c 'identifier', ident
+
+  keyword = c 'keyword', word {
+    'break', 'case', 'chan', 'const', 'continue', 'default', 'defer', 'else',
+    'fallthrough', 'for', 'func', 'go', 'goto', 'if', 'import', 'interface',
+    'map', 'package', 'range', 'return', 'select', 'struct', 'switch', 'type',
+    'var'
+  }
+
+  type = c 'type', word {
+    'bool', 'byte', 'complex64', 'complex128', 'float32', 'float64', 'int8',
+    'int16', 'int32', 'int64', 'rune', 'string', 'uint8', 'uint16', 'uint32', 'uint64',
+    'complex', 'float', 'int', 'uint', 'uintptr'
+  }
+  
+  builtin = c 'function', word {
+    'append', 'cap', 'close', 'complex', 'copy', 'delete', 'imag', 'len', 'make',
+    'new', 'panic', 'print', 'println', 'real', 'recover'
+  }
+
+  fdecl = sequence {
+    c 'keyword', 'func'
+    any {
+      sequence({
+        ws^0
+        c 'operator', '('
+        (P 1 + ws - P')')^0
```

Comment:
So this would lex everything inside as any symbol? Not sure this makes
sense.

On Mon, 29 Feb 2016 at 17:21 Ryan Gonzalez <notifications@github.com> wrote:

> In bundles/go/go_lexer.moon
> <https://github.com/howl-editor/howl/pull/175#discussion_r54443790>:
>
> > +    'int16', 'int32', 'int64', 'rune', 'string', 'uint8', 'uint16', 'uint32', 'uint64',
> > +    'complex', 'float', 'int', 'uint', 'uintptr'
> > +  }
> > +
> > +  builtin = c 'function', word {
> > +    'append', 'cap', 'close', 'complex', 'copy', 'delete', 'imag', 'len', 'make',
> > +    'new', 'panic', 'print', 'println', 'real', 'recover'
> > +  }
> > +
> > +  fdecl = sequence {
> > +    c 'keyword', 'func'
> > +    any {
> > +      sequence({
> > +        ws^0
> > +        c 'operator', '('
> > +        (P 1 + ws - P')')^0
>
> @barnybug <https://github.com/barnybug> Sorry, I'm an idiot. :/ The
> reason the version I gave didn't work was because this line is supposed to
> be ((V'all' + P 1 + ws) - P')')^0 (note the extra parens).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/howl-editor/howl/pull/175/files#r54443790>.
>